src/promptflow/promptflow/_core/tools_manager.py

```diff
@@ -277,9 +280,16 @@ def _load_tool_from_module(
             if k not in init_inputs:
                 continue
             if v.value_type != InputValueType.LITERAL:
-                raise ValueError(
-                    f"Input {k!r} for tool '{tool_name}' only supports literal values for initialization,"
-                    + f" got {v.serialize()!r}"
+                raise InputTypeMismatch(
+                    message_format=(
+                        "The input for flow is incorrect. The tool '{tool_name}' requires a literal value for "
+                        "the input '{input_name}', however, a value '{input_value}' was provided which is not a "
+                        "literal value. Please provide a literal value for '{input_name}' to correct this issue."
+                    ),
```

Comment:
Let's try to refine the error message to remove some duplicated parts like `{input_name}`.

Reply:
Fixed.